From sarah.a.sharp@linux.intel.com  Thu Apr 29 12:57:38 2010
From: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Date: Fri, 2 Apr 2010 15:33:56 -0700
Subject: USB: Add parsing of SuperSpeed endpoint companion descriptor.
To: Greg KH <gregkh@suse.de>
Cc: linux-usb@vger.kernel.org, usb-storage@lists.one-eyed-alien.net, Hrant Dalalyan <Hrant.Dalalyan@synopsys.com>, Alan Stern <stern@rowland.harvard.edu>, Paul Zimmerman <Paul.Zimmerman@synopsys.com>, Ashot Madatyan <Ashot.Madatyan@synopsys.com>
Message-ID: <20100402223356.GA1235@xanatos>
Content-Disposition: inline


Allow the xHCI drivers (and any new USB 3.0 drivers) to parse the
SuperSpeed endpoint companion descriptor to find the maximum number of
bulk endpoint streams the endpoint supports.  This is used to calculate
the maximum total number of streams the driver can allocate.

Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/usb/ch9.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/usb/ch9.h
+++ b/include/linux/usb/ch9.h
@@ -556,6 +556,8 @@ struct usb_ss_ep_comp_descriptor {
 } __attribute__ ((packed));
 
 #define USB_DT_SS_EP_COMP_SIZE		6
+/* Bits 4:0 of bmAttributes if this is a bulk endpoint */
+#define USB_SS_MAX_STREAMS(p)		(1 << (p & 0x1f))
 
 /*-------------------------------------------------------------------------*/
 
